python/spear/sp_engine.py

```diff
@@ -0,0 +1,271 @@
+            elif sys.platform == "darwin":
+                paks_dir = \
+                    os.path.realpath(os.path.join(self._config.SPEAR.STANDALONE_EXECUTABLE, "Contents", "UE", "SpearSim", "Content", "Paks"))
+            elif sys.platform == "linux":
+                paks_dir = \
+                    os.path.realpath(os.path.join(os.path.dirname(os.path.realpath(self._config.SPEAR.STANDALONE_EXECUTABLE)), "SpearSim", "Content", "Paks"))
+            else:
+                assert False
+
+            assert os.path.exists(paks_dir)
+
+            # we don't use os.path.realpath here because we don't want to resolve the symlink
+            spear_paks_dir = os.path.join(paks_dir, "SpearPaks")
+
+            if spear.path_exists(spear_paks_dir):
+                spear.log(f"File or directory or symlink exists, removing: {spear_paks_dir}")
+                spear.remove_path(spear_paks_dir)
+
+            spear.log(f"Creating symlink: {spear_paks_dir} -> {self._config.SPEAR.PAKS_DIR}")
+            os.symlink(self._config.SPEAR.PAKS_DIR, spear_paks_dir)
+
+        # provide additional control over which Vulkan devices are recognized by Unreal
+        if self._config.SPEAR.VK_ICD_FILENAMES != "":
+            spear.log("Setting VK_ICD_FILENAMES environment variable: " + self._config.SPEAR.VK_ICD_FILENAMES)
+            os.environ["VK_ICD_FILENAMES"] = self._config.SPEAR.VK_ICD_FILENAMES
+
+        # set up launch executable and command-line arguments
+        launch_args = []
+
+        if self._config.SPEAR.LAUNCH_MODE == "uproject":
+            launch_executable = self._config.SPEAR.UNREAL_EDITOR_EXECUTABLE
+            launch_args.append(self._config.SPEAR.UPROJECT)
+            launch_args.append("-game") # launch the game using uncooked content
+        elif self._config.SPEAR.LAUNCH_MODE == "standalone_executable":
+            launch_executable = self._config.SPEAR.STANDALONE_EXECUTABLE
+        else:
+            assert False
+
+        assert os.path.exists(launch_executable)
+
+        launch_executable_name, launch_executable_ext = os.path.splitext(launch_executable)
+
+        if sys.platform == "win32":
+            assert launch_executable_name[-4:] == "-Cmd"
+            assert launch_executable_ext == ".exe"
+            launch_executable_internal = launch_executable
+        elif sys.platform == "darwin":
+            assert launch_executable_ext == ".app"
+            launch_executable_internal = os.path.realpath(os.path.join(launch_executable, "Contents", "MacOS", os.path.basename(launch_executable_name)))
+        elif sys.platform == "linux":
+            assert launch_executable_ext == ".sh"
+            launch_executable_internal = launch_executable
+        else:
+            assert False
+
+        assert os.path.exists(launch_executable_internal)
+
+        launch_args.append("-windowed")
```

Comment:
Move all command-line args to a YAML dictionary that can be customized by the user. If a command-line arg is just a value, i.e., doesn't need an equals sign, then it's value in the YAML dictionary should be `null`.